sfcasts/ep1/computed.md

```diff
@@ -1,228 +1,192 @@
-
+Here, PHPStorm *tries* to guess the return type... but gets a little confused
+since `vue` is *so* dynamic. Let's help it out: this returns an array of strings.
 
 Excellent!
 
 ## How computed works
- 
-So here's the deal. As soon as you have a key under the `computed` object, this
-becomes available in the template as a variable. In fact, `vue` will create a
-variable with a `getter` set to our method and assign it to its magical `this` keyword.
-But most importantly, if *anything* inside that method changes, weather it be a
-`data` property or a `prop`, `Vue` will automatically re-run that method for us
-and recalculate the rendered DOM in a *very* optimized way. *Wow!* Your world
-just changed forever!
-
-So we'll copy `componentClass` and up here in the template, very simply we'll say
-`:class="componentClass"`. I can also organize this better and move our `<div>`
-on a single line. That's it, ESlint, don't be mad at me!
- 
-So up until now, we know that `vue` adds any `data` `props` or `methods` to the vue
-instance, which means that we can reference those inside of our template.
-Well, `computed` is the fourth and final thing that gets added to that instance
-and as we just said, `Vue` adds any key under `computed` as a `property` with its
-`getter` set to that method.
-
-So in up here back in the template, we can just reference `componentClass`,
-which is really `this.componentClass`. And behind the scenes `vue` is going to
-call this method `componentClass` to get that and update it whenever necessary.
-
-Because this is being added to the `vue` instance, just like `methods`, `data` or
-`props`, it means that we can actually reference it with the `this` variable.
-Just to prove it, inside of our `toggleCollapsed` method, let's say
-`console.log(this.componentClass())`.
- 
- 
- 
-Notice that PHPStorm tries to auto complete that with parentheses, but that's not
-correct! We need to reference it like a property, even though behind the scenes,
-we know view is going to call this method.
+
+So here's the deal. As soon as you have a key under the `computed` option, it
+becomes available in the template as a *variable*.
+
+Copy `componentClass` and, up in the template, very simply, we'll say
+`:class="componentClass"`.
+
+So up until now, we know that Vue adds any `data`, `props` or `methods` to the Vue
+instance, which means that we can reference those inside our template.
+Well, `computed` is the fourth and *final* thing that gets added to the instance.
+`Vue` adds each key under `computed` as a *property*.
+
+This means that, up in the template, we can just reference `componentClass`,
+which is really `this.componentClass`. But behind the scenes, when we access that
+property, Vue will *actually* call the `componentClass` method to get it. It's able
+to do that thanks to the fake getter properties we saw earlier.
+
+And really, the only difference between `methods` and `computed` is the syntax: we
```

Comment:
I'm not sure how I feel about it. Really methods and computed are 2 very different things! You need to `call` a method to get its code to run, where a `computed property` will call itself when it needs to. May be there's a way to squeeze the concept in here!

Reply:
Hmm. From my perspective, the difference is nothing but syntactic sugar + the caching part. You DO need to call a method to get its code to run. The same is true for a computed property, you just use `this.theProperty` to effectively "call" the method. In both cases, when your template is being rendered, Vue follows your template and calls methods & calls computed properties. Of course, if you reference a computed property 2 times in a render call, it's smart enough not to call your function twice. 

But I think what I said here is true. You can even use/abuse a method to return a value. And if you access some data/props in that method, those are added to the Observer's "dependency" list so that when that data/prop changes, the component re-renders.

But maybe I'm unaware of a deeper difference? Let me know :)

Reply:
No, no method is called unless to explicitly call it. Setting @click="myMethod" won't call that method. Also, if a method has data/props in it, changing those data/props, won't call the method again. You need to call it yourself.

Also, computed properties aren't methods at all! Your "method" to determine the value is wrapped into a getter, which is a method, but not one you can "call"